### PR TITLE
Cloud 1.0.1

### PR DESCRIPTION
**Pull request by natedogs911**
aims to close Open-Network-Insight/open-network-insight#22

this employs the AWS CLI and Python BOTO3 to do the following

simple bash script (aws_to_s3.sh) pushes binary files to s3://bucket/staging
ingest framework uses BOTO3 to watch s3://bucket/staging for incoming files, cp file to local fs staging folder and mv the file from s3://bucket/staging--> s3://bucket/archive bucket
notification is sent via rabbitMQ and workers consume the file from local fs

### CODE REVIEW

File: ingest/dns_to_s3.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
```
Comment (NathanSegerlind) [inline ingest/dns_to_s3.sh:1]: could you add a couple lines of comments detailing what this file does?
maybe just sentence...

File: ingest/dns_to_s3.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+#####################################
+# please customize the below fields #
```
Comment (NathanSegerlind) [inline ingest/dns_to_s3.sh:4]: how about treat them as arguments with default values if not provided?

File: ingest/etc/master_ingest.json [hunk 3]
```diff
@@ -1,17 +1,23 @@
 {
     "dns":{
             "huser" : "/user/oni",
-            "collector_path":"/mnt/sec_shared.nfs/dns",
+            "collector_path":"/mnt/ebs/dns",
```
Comment (NathanSegerlind) [inline ingest/etc/master_ingest.json:4]: does this change how all ingest is handled now?

File: ingest/flow_to_s3.sh [hunk 4]
```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
```
Comment (NathanSegerlind) [inline ingest/flow_to_s3.sh:1]: same comments as for the dns_to_s3 script.... 

the script should have a comment at top saying what it does, and expecting people to simply edit values (rather than use CLI parameters) strikes me as sketchy but maybe the bash script world is different

File: ingest/oni/dns_master.py [hunk 5]
```diff
@@ -3,107 +3,127 @@
...
+        self._dsource = 'dns'
+
+        # valdiate configuration info.
+        conf_err_msg = "Please provide a valid '{0}' in the configuration file"
+        Util.validate_parameter(conf['collector_path'], conf_err_msg.format("collector_path"))
+        Util.validate_parameter(conf['queue_name'], conf_err_msg.format("queue_name"))
+        Util.validate_parameter(conf['pkt_num'], conf_err_msg.format("pkt_num"))
+        Util.validate_parameter(conf['pcap_split_staging'], conf_err_msg.format("pcap_split_staging"))
+        Util.validate_parameter(conf['time_to_wait'], conf_err_msg.format("time_to_wait"))
+        #Util.validate_parameter(conf['s3Bucket'], conf_err_msg.format("s3Bucket"))
```
Comment (NathanSegerlind) [inline ingest/oni/dns_master.py:36]: no reason to keep around code in the comments

code in the comments doesn't get tested or updated as changes come in... eventually it will rot and break the rest of the program if it is uncommented

it also confuses the reader (why is this here but commented out?)

File: ingest/oni/dns_master.py [hunk 6]
```diff
@@ -3,107 +3,127 @@
...
+
+        # get file size.
+        file_size = os.stat(file_local_path)
+        if file_size.st_size > 1145498644:
+
+            # split file.
+            self._split_pcap_file(file_name, file_local_path, local_path)
+        else:
+
+            # send rabbitmq notification.
```
Comment (NathanSegerlind) [inline ingest/oni/dns_master.py:102]: this comment is redundant with the method name below (or there is an important distinction about notifying RabbitMQ that is not suggested in the method call)

File: ingest/oni/dns_master.py [hunk 7]
```diff
@@ -3,107 +3,127 @@
...
+            #local_pcap_file = "{0}/{1}".format(local_path, file_name)
```
Comment (NathanSegerlind) [inline ingest/oni/dns_master.py:103]: delete commented out code

File: ingest/oni/load_dns_avro_parquet.hql [hunk 8]
```diff
@@ -11,18 +11,18 @@ DROP TABLE IF EXISTS ${hiveconf:dbname}.dns_tmp
```
Comment (NathanSegerlind) [inline ingest/oni/load_dns_avro_parquet.hql:12]: are these changes part of your changes or simply changes that you merged in?

File: ingest/oni/utils.py [hunk 9]
```diff
@@ -1,77 +1,70 @@
...
+        channel.queue_declare(queue=queue_name)
+        channel.basic_publish(exchange='', routing_key=queue_name, body=file)
+        connection_send.close()
+
+    @classmethod
+    def build_hdfs_path(cls, file, data_type):
+
+        if data_type == 'flow':
+
+            # get file name and date.
```
Comment (NathanSegerlind) [inline ingest/oni/utils.py:44]: comment has gone stale since getting file name has been commented out

delete all three comments, they add nothing

**Comment by natedogs911** [comment]
i will be submitting a new version to work with kafka soon
